src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -205,13 +210,26 @@ public DatasetTerm toDatasetTerm(Dataset dataset) {
     DatasetTerm term = new DatasetTerm();
 
     term.setDatasetId(dataset.getDataSetId());
+    if (Objects.nonNull(dataset.getCreateUserId())) {
+      User user = userDAO.findUserById(dataset.getCreateUserId());
+      term.setCreateUserId(dataset.getCreateUserId());
+      term.setCreateUserDisplayName(user.getDisplayName());
```

Comment:
I'm guessing it's OK to do this because a user's display name can't change over time? Or is there a step that recreates the index when a field like this changes?

Reply:
This code is essentially the place where we recreate the index. There are some hooks that call this when a DUOS dataset is changed or when a DAR is approved/denied - all of those hooks flow through this method. We don't allow for users to change their display name, however, so that's always going to be whatever it was when they originally signed in.